src/components/action-bar/action-bar-item/action-bar-item.tsx

```diff
@@ -0,0 +1,134 @@
+import { Component, Event, EventEmitter, h, Prop } from '@stencil/core';
+import { ActionBarItem } from '../action-bar.types';
+import { createRandomString } from '../../../util/random-string';
+import { ListSeparator } from 'src/interface';
+
+/**
+ * @private
+ */
+@Component({
+    tag: 'limel-action-bar-item',
+    shadow: false,
+    styleUrl: 'action-bar-item.scss',
+})
+export class ActionBarButton {
+    /**
+     * Item that is placed in the action bar.
+     */
+    @Prop()
+    public item!: ActionBarItem | ListSeparator;
+
+    /**
+     * Fired when a action bar item has been clicked.
+     */
+    @Event()
+    public select: EventEmitter<ActionBarItem | ListSeparator>;
+
+    /**
+     * When the item is displayed in the available width,
+     * this will be `false`.
+     */
+    @Prop({ reflect: true })
+    public isVisible: boolean = true;
```

Comment:
this strategy gives us many things, but it also creates empty wholes in the UI like this:
![image](https://github.com/Lundalogik/lime-elements/assets/35954987/ff091a86-665a-44f1-aeef-0752754e6ea6)
@LawrenceBorst and I are trying to see how we can mitigate this.

Reply:
I think the current behavior 👇 

https://github.com/Lundalogik/lime-elements/assets/35954987/ff684a7e-f5e0-4738-8757-e96cb53ccc25

is still better than the alternative _jumping_ solution that we can do now 👇 

https://github.com/Lundalogik/lime-elements/assets/35954987/79257559-db76-43a7-b9e7-4523334c8a2e

I'm still not sure how we can achieve a nice and smooth transition, when an item disappears. We don't want tit to occupy visible space anymore. I can think of two ways to do it with CSS, one would be setting `display: none` on it, and the other is using `position: absolute` on the `<button />`. But both of these will mess up the intersection observer. Which means the observer never puts back the "removed" item, because it can't see it anymore.

I think we can skip fixing this issue for now. Maybe some beautiful day we can make it work using more advanced CSS and JS.

Reply:
While I agree with Lucy to some extent, I don't think this is about handling resizing as much as it's about handling re-rendering and having something that's not too complex and not shaky.

I think we are happy with the empty hole for now, so that this can be merged? And hopefully we'll find a way to improve it.

Reply:
I would say let's be fine with the empty hole. 🕳️ 
If the apatite grows to fill it we could do it in a cooldown.